rfcs/2021-28-07-motion-design-core-actions.md

```diff
@@ -0,0 +1,176 @@
+1. What unit we'll be using for these core actions defined?
+2. How will we ensure accessibility in terms of motion?
+3. How will we use platform dependent units(px, rem, dp etc) in order to render the animation on a particular platform in a generic way without polluting our design system components.
+4. How/where we will store the unit as design tokens?
+
+# Detailed Design
+## Core Actions
+These are all the actions that can affect an element while implementing motion on an element:
+1. **Scale:** Change in the dimensions; it can also be represented as moving in `z-axis` if `x` & `y` axes are constant.
+2. **Move:** Change in the positions; generally in `x` & `y` axes.
+3. **Fade:** Change in the opacity, appearing or disappearing of the element.
+4. **Rotation:** Change in the rotation on `z-axis`.
+5. **Fill:** Change in colour.
+6. **Flip:** Change in the rotation on `x-axis`.
+7. **Morph:** Change in the shape.
+8. **Extrude:** Change in the thickness.
+9. **Skew:** Change in the angle between sides of a polygon.
+10. **Stroke size:** change in the width of borders.
+11. **Blur:** Change in the sharpness; such as shadows.
+12. **Trim Path:** Change in the start-point and end-point of a line.
+
+## Animation Properties of the Core Actions
+Based on our the products pre-existing actions and branding we've decided on these 5 core-actions and their attributes:
+1. **Scale:** `length`, `height`, `anchor-point`, `size`.
+2. **Move:** `anchor-points`, `pathway`.
+3. **Fade:** `opacity`.
+4. **Fill:** `colour-tokens`.
+5. **Morph:** `shape-attributes`.
+
+## Atrribute Details
+### 1. **Scale**
+* **Length:** Size on `x-axis`.
+* **Height:** Size on `y-axis`.
+* **Anchor point:** Coordinates of a point in the UI element which define the position of the element; usually it’s the center of the element. Basically a point in which all transformations are manipulated from.
+* **Size:** Percentage(`%`) change in size if aspect ratio is constant.
+
+### Types:
+| Type | Description | Example |
+|---|---|---|
+| **Zoom** | Where the `aspect-ratio` is the same; don’t squash and stretch UI elements<br>**Ex:** Hovering a card.<br>**Unit:** `%` | <img src="images/motion/core-actions/image4.gif" alt="core-action-scale-zoom" width="200px"> |
+| **Expansion** | Where `length` and `height` changes separately.<br>**Ex:** Expansion of a modal or opening of a dropdown.<br>**Unit:** `px`,`%`,`%y`,`%` | <img src="images/motion/core-actions/image5.gif" alt="core-action-scale-expand" width="200px"> |
+
+### 2. **Move**
+* **Anchor points(`I` & `F`):** Initial & Final anchor-point coordinates; `(x1, y1)(x2, y2)`.
+* **Pathway:** Control points (which decide curve of the line) [path](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths) i.e., how the element moves from `position1` to `position2` representing the pathways; depending on the surroundings or the `z-axis`.
+
+
+### Pathway Types:
```

Comment:
Not able to understand pathway types. Maybe need to discuss on a call

Reply:
https://material.io/design/motion/customization.html#motion-paths
Check out the link. I will explain on call.